nixos/modules/services/x11/xserver.nix

```diff
@@ -313,12 +335,27 @@ in
 
       xrandrHeads = mkOption {
         default = [];
-        example = [ "HDMI-0" "DVI-0" ];
-        type = with types; listOf string;
+        example = [ "HDMI-0" { output = "DVI-0"; primary = true; monitorConfig = ""; } ];
+        type = with types; listOf (either
+          str
+          (submodule {
+            options.output = str;
+            options.primary = bool;
+            options.monitorConfig = lines;
```

Comment:
@aszlig is the only advantage is to have a description field? Does that go into the auto generated documentation? Also the `default` inside the submodules doesn't work!